### ISSUE TITLE
TypeError: ledgerPublisher.getMedia.getPublisherFromMediaProps is not a function (entire UI becomes unresponsive)

### ISSUE THREAD

**Issue by kjozwiak** [brave/browser-laptop#12096]
### Description

The entire Brave UI becomes unresponsive when watching YT media while the wallet is enabled. You'll also see the following errors:

```
An uncaught exception occurred in the main process Uncaught Exception:
TypeError: ledgerPublisher.getMedia.getPublisherFromMediaProps is not a function
    at Object.onMediaRequest (/Applications/Brave.app/Contents/Resources/app.asar/app/browser/api/ledger.js:2303:28)
    at ledgerReducer (/Applications/Brave.app/Contents/Resources/app.asar/app/browser/reducers/ledgerReducer.js:426:27)
    at reducers.reduce (/Applications/Brave.app/Contents/Resources/app.asar/js/stores/appStore.js:383:24)
    at Array.reduce (<anonymous>)
    at applyReducers (/Applications/Brave.app/Contents/Resources/app.asar/js/stores/appStore.js:381:68)
    at handleAppAction (/Applications/Brave.app/Contents/Resources/app.asar/js/stores/appStore.js:440:16)
    at callbacks.forEach (/Applications/Brave.app/Contents/Resources/app.asar/js/dispatcher/appDispatcher.js:107:7)
    at Array.forEach (<anonymous>)
    at AppDispatcher.dispatchToOwnRegisteredCallbacks (/Applications/Brave.app/Contents/Resources/app.asar/js/dispatcher/appDispatcher.js:106:20)
    at AppDispatcher.dispatchInternal (/Applications/Brave.app/Contents/Resources/app.asar/js/dispatcher/appDispatcher.js:132:10)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
```

I'm assuming there's an issue when Brave tries to add YT entries into the ledger.

### Steps to Reproduce

  1. launch `0.19.96` (UGP)
  2. open a new tab and enable payments via `about:preferences#payments`
  3. once the wallet has been enabled, select "Maybe Later" under the UGP notification
  4. open a new tab and load YT
  5. watch about ~1min of any random video and you'll notice the entire brave UI will freeze including the error mentioned above.

**Actual result:**

The entire Brave UI becomes unresponsive when watching YT and the wallet has been enabled.

Example of issue: https://youtu.be/yOJu9KS4Nfs

**Expected result:**

The entire Brave UI shouldn't be unresponsive when watching YT... The above `TypeError` shouldn't be occurring either.

**Reproduces how often:**

100% reproducible using the above STR.

### Brave Version

**about:brave info:**

>  Brave: 0.19.96 
> rev: 9d7294456e183a02eb31f53a4cda40fdd51acb24 
> Muon: 4.5.16 
> libchromiumcontent: 62.0.3202.94 

**Reproducible on current live release:**

No, above issue is not reproducible on `0.19.95`. I'm sure we would have heard from our community by now if this was in the release build.

### Additional Information

* macOS 10.12.6 x64 - Reproduced
* Win 10 Pro x64 - Reproduced
* Ubuntu 17.19 x64 - Reproduced

**Comment by srirambv** [comment]
Unable to reproduce on Windows x64 on a clean profile. But I do see the same error in console but UI is still  usable. Although YT publisher is not added to the ledger entry only YouTube is added with 17s visit time

**Comment by NejcZdovc** [comment]
this was broken with update of `bat-publishers` to 2.x.x

**Comment by NejcZdovc** [comment]
fixed with 23f6d2f15
